icu4c/source/common/dictbe.cpp

```diff
@@ -325,7 +325,7 @@ ThaiBreakEngine::divideUpDictionaryRange( UText *text,
                         // two characters after uc were not 0x0E4C THANTHAKHAT before
                         // checking the dictionary. That is just a performance filter,
                         // but it's not clear it's faster than checking the trie.
-                        int32_t candidates = words[(wordsFound + 1) % THAI_LOOKAHEAD].candidates(text, fDictionary, rangeEnd);
+                        candidates = words[(wordsFound + 1) % THAI_LOOKAHEAD].candidates(text, fDictionary, rangeEnd);
```

Comment:
What was the warning here? Going from a small scope to a large scope seems possibly risky. (Might be fine but I don't have the context)

Reply:
The warning was "warning C4456: declaration of 'candidates' hides previous local declaration", but you're right I should be changing the variable name instead of changing the scope.